buildengine/engine.go

```diff
@@ -189,6 +189,9 @@ func (e *Engine) Graph(projects ...ProjectKey) (map[string][]string, error) {
 
 func (e *Engine) buildGraph(key string, out map[string][]string) error {
 	var deps []string
+	if _, ok := out[key]; ok {
+		return fmt.Errorf("module %q contains a cyclical dependency", key)
```

Comment:
🎉

Reply:
turns out this is a little too simplistic of an approach - I'm going to iterate on this more. Because `buildGraph` explores fully resolved dependencies (such as `builtin`) this will result in a ton of false positives

Reply:
updated to move the cycle detection code to the topological sort function.`buildGraph` now simply short circuits when encountering a modules that has already visited